### PR TITLE
feat: add device, logger and poll

### PR DESCRIPTION
**Pull request by stayradiated**
This code has been copied over from Ghostphone, and converted to typescript.

### CODE REVIEW

File: src/device/index.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    const devices = await client.listDevices()
+    if (devices.length < 1) {
+      throw new Error('Could not find any devices :(')
+    }
+    const deviceId = devices[0].id
+    return new Device(deviceId)
+  }
+
+  constructor (deviceId: string) {
+    this.id = deviceId
```
Comment (jladuval) [inline src/device/index.ts]: This confuses me a bit, so the only reason this is a class is to hold the device id? Could we not just inject the ID into most of these functions then we could test them a lot easier?

Comment (stayradiated) [reply src/device/index.ts]: Yep, the use of a class is just a style choice. It doesn't make much difference to testing.

```typescript
import { Device } from '@mishguru/muppet'

const device = await Device.firstDeviceAvailable()
await device.pullFile('/sdcard/window_dump.xml', destPath)
```

vs

```typescript
import { device } from '@mishguru/muppet'

const deviceId = await device.firstDeviceIdAvailable()
await device.pullFile(deviceId, '/sdcard/window_dump.xml', destPath)
```

Comment (jladuval) [reply src/device/index.ts]: I'd prefer to move these out of a class for maintainability sake. It's only one field and if we don't do it now we probably never will

Comment (stayradiated) [reply src/device/index.ts]: Ok, I'll do it tomorrow

**Review by jladuval — APPROVED** [review]

**Pull request merged**